### DISCUSSION TITLE
Porting a mod of a mod of Doom to WebAssembly

### DISCUSSION THREAD

**Discussion by giscus** [SkyLeite/blog#8] [Announcements]
# Porting a mod of a mod of Doom to WebAssembly

https://leite.dev/

**Comment by ayunami2000** [comment]
Hey! I wanted to see if I could work with you on a WebSocket or WebRTC implementation, possibly exclusive to this port. Dm me on Discord: ayunami2000 (ayunami2000#5250) plz!! Thanks!

**Comment by SkyaTura** [comment]
Not to mention that Doom can also run Doom haha

**Reply by SkyLeite** [reply]
We have to go deeper

**Reply by SkyaTura** [reply]
That's it!

It's time to create a JS Runtime in Doom!

**Comment by pizzatowerfan126** [comment]
can you make it so mods can work?

**Reply by SkyLeite** [reply]
Hey! They should already be working. All you have to do is click the Add Addon button at the bottom and select a mod file from your computer.

**Reply by pizzatowerfan126** [reply]
i meant for the non kart one, also can you make the version for it 2.2.13 pls?

**Reply by SkyLeite** [reply]
I see. To be honest I don't really have the bandwidth at the moment, and updating to a new version would essentially entail doing the work all over again from scratch. Not to say it's not possible, but it would take more time than I have now :/ I'll let you know if that changes though, or if someone from the community feels like taking up on the challenge.

**Reply by EladioPalacios** [reply]
How about you just leave the version you've already made, but just make mods work? What I mean is just leave the version you used. (2.2.11) and just try if you can allow it to run mods! Another good change could be saving! If it's easier for you, you can add a option to just download saves, that then you don't need to do all that local save in browser thing!

**Reply by SkyaTura** [reply]
Please consider that developers also have bills to pay, and jobs to attend in order to pay them. Unfortunately, as much as we like to make cool projects like this, what we usually do in work hours is not as cool. Matter in fact, working full time is exaustive AF, and in the remaining hours of the day we don't have any energy left to do things like this.

The great thing about open projects is the fact _other people_ interested on making this changes can do it if they want to. I do believe your intentions are the best possible, but as a developer myself, I find needed to say that  this kind of pressure against something that was already explained very discouraging on creating new things in the future.

**Reply by SkyLeite** [reply]
I think those are good ideas. Adding mod support and uploading / downloading save files sounds pretty reasonable. I don't know when I'll have the bandwidth for that, but it's definitely much less effort than updating to a new version of the game. Thanks for the suggestion :)

**Reply by EladioPalacios** [reply]
It's fine! Take the time you need, so don't worry! I'll stick with Vin's SRB2 web port! It may not run the latest mods, but it's fun still!

**Reply by EladioPalacios** [reply]
How's it going so far

**Reply by pizzatowerfan126** [reply]
a sequel to srb2k was released.

**Comment by EladioPalacios** [comment]
Well, we might as well use other methods, but I get you.

**Comment by EladioPalacios** [comment]
What is it called?

**Reply by pizzatowerfan126** [reply]
Dr. Robotnik's Ring Racers

**Comment by EladioPalacios** [comment]
Okay! Thanks! Hope there are mods for it in the future. (Even though it's
probably a demo for now)

**Comment by pizzatowerfan126** [comment]
I'm thinking about porting ring racers to WebAssembly but i know NOTHING about it and im only in 7th grade.

**Comment by pizzatowerfan126** [comment]
does anybody know how to make a deployment on github?

**Reply by pizzatowerfan126** [reply]
because I'm trying to port ring racers to web (and I need someone to teach me).

**Reply by EladioPalacios** [reply]
YOU CAN NOW USE MODS IN SRB2, I REPEAT. YOU CAN NOW RUN MODS IN SKY'S SRB2 WEB PORT (Still you can't save but IDC MODS YAY)
